tests/tracer/test_logging.py

```diff
@@ -0,0 +1,202 @@
+import os
+
+
+def test_unrelated_logger_in_debug_with_patch_all(run_python_code_in_subprocess):
+    """
+    When importing the tracer after the custom log,
+                the ddtrace logger does not override any custom logs settings
+    """
+    env = os.environ.copy()
+    env["DD_TRACE_DEBUG"] = "true"
+    env["DD_TRACE_LOG_FILE"] = "./testlog.log"
+    code = """
+import logging
+custom_logger = logging.getLogger('custom')
+custom_logger.setLevel(logging.WARN)
+
+assert custom_logger.parent.name == 'root'
+assert custom_logger.level == logging.WARN
+
+import ddtrace
+
+ddtrace.patch_all()
+
+assert custom_logger.parent.name == 'root'
+assert custom_logger.level == logging.WARN
+"""
+
+    out, err, status, pid = run_python_code_in_subprocess(code, env=env)
+    assert status == 0, err
+    assert err == b""
+
+
+def test_unrelated_logger_loaded_last_patch_all(run_python_code_in_subprocess):
+    """
+    When importing the tracer before the custom log,
+                the ddtrace logger does not override any custom logs settings
+    """
+    env = os.environ.copy()
+    env["DD_TRACE_DEBUG"] = "true"
+    env["DD_TRACE_LOG_FILE"] = "./testlog.log"
+    code = """
+import ddtrace
+import logging
+
+ddtrace.patch_all()
+
+custom_logger = logging.getLogger('custom')
+custom_logger.setLevel(logging.WARN)
+
+assert custom_logger.parent.name == 'root'
+assert custom_logger.level == logging.WARN
+"""
+
+    out, err, status, pid = run_python_code_in_subprocess(code, env=env)
+    assert status == 0, err
+    assert err == b""
+
+
+def test_child_logger_inherits_settings_patch_all(run_python_code_in_subprocess):
+    """
+    When setting up a child logger under ddtrace, it inherits from ddtrace automatically.
+    """
+    env = os.environ.copy()
+    env["DD_TRACE_DEBUG"] = "true"
+    env["DD_TRACE_LOG_FILE"] = "./testlog.log"
+    code = """
+import logging
+from ddtrace.logger import configure_ddtrace_logger
+import ddtrace
+
+ddtrace.patch_all()
+
+child_logger = logging.getLogger('ddtrace.child')
+assert child_logger.parent.name == 'ddtrace'
+assert child_logger.parent.getEffectiveLevel() == child_logger.getEffectiveLevel()
+assert child_logger.getEffectiveLevel() == logging.DEBUG
+assert child_logger.handlers == []
```

Comment:
For this test case, we don't need to assert that `ddtrace.child` is setup correctly. We can assume that Python's standard library will properly setup child loggers.

Instead we can have this assert that the main `ddtrace` logger is setup as expected either from just importing `ddtrace` or with `ddtrace-run`

Reply:
Does that mean `def test_child_logger_inherits_settings(run_python_code_in_subprocess, tmpdir):` as a test isn't needed?

Reply:
I removed `test_child_logger_inherits_settings` as a test.